packages/actor-extract-links-extract-tree/lib/treeMetadataExtraction.ts

```diff
@@ -0,0 +1,135 @@
+import type { IRelation, IRelationDescription } from '@comunica/types-link-traversal';
+import { RelationOperator, TreeNodes } from '@comunica/types-link-traversal';
+import type * as RDF from 'rdf-js';
+
+export function collectRelation(
+  relationDescription: IRelationDescription,
+  nodeLinks: string,
+): IRelation {
+  const relation: IRelation = { node: nodeLinks };
+  const typeRelation = (() => {
+    if (typeof relationDescription.operator !== 'undefined') {
+      return typeof relationDescription.operator[0] !== 'undefined' ? relationDescription.operator[0] : undefined;
+    }
+  })();
+
+  if (typeof typeRelation !== 'undefined' && typeof relationDescription.operator !== 'undefined') {
```

Comment:
```suggestion
  if (typeRelation && relationDescription.operator) {
```
Also in other parts of the code

Reply:
^This would make the code a lot more readable :-)